analytics/endpoint_data_objects.go

```diff
@@ -0,0 +1,88 @@
+package analytics
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+)
+
+type RequestType string
+
+const (
+	COOKIE_SYNC RequestType = "/cookie_sync"
+	AUCTION     RequestType = "/openrtb2/auction"
+	SETUID      RequestType = "/set_uid"
+	AMP         RequestType = "/openrtb2/amp"
+)
+
+//Loggable object of a transaction at /openrtb2/auction endpoint
+type AuctionObject struct {
+	Type      RequestType
+	Status    int
+	Error     []error
+	Request   *openrtb.BidRequest
+	Response  *openrtb.BidResponse
+	UserAgent string
+}
+
+//Loggable object of a transaction at /openrtb2/amp endpoint
+type AmpObject struct {
+	Type            RequestType
+	Status          int
+	Error           []error
+	Request         *openrtb.BidRequest
+	AuctionResponse *openrtb.BidResponse
+	AmpResponse     map[string]string
+	UserAgent       string
+	Origin          string
```

Comment:
Wouldn't `UserAgent` and `Origin` be found in `BidRequest`? Is there a reason these are being sent externally?

Reply:
I don't think they are in the BidRequest..

Reply:
The `UserAgent` is `request.device.ua`. Not sure about the `Origin`.

(I'll do an actual review of this tomorrow)